### PR TITLE
[IMP] spreadsheet: handle locales in spreadsheet

### PR DESCRIPTION
**Pull request by hokolomopo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

Task: [3222401](https://www.odoo.com/web#id=3222401&menu_id=4720&cids=1&action=333&active_id=2328&model=project.task&view_type=form)

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/utils.py [hunk 1]
```diff
@@ -1,26 +1,97 @@
 import json
 import base64
+import re
 
 from odoo import _
 
-def empty_spreadsheet_data_base64():
+def empty_spreadsheet_data_base64(lang):
```
Comment (LucasLefevre) [inline addons/spreadsheet/utils.py:7]: you can move `empty_spreadsheet_data_base64` and `empty_spreadsheet_data` to `spreadsheet.mixin` methods.
That way you can remove the parameter and all those `_get_lang` before every call

File: addons/spreadsheet/utils.py [hunk 2]
```diff
@@ -1,26 +1,97 @@
...
+    data = json.dumps(empty_spreadsheet_data(lang))
     return base64.b64encode(data.encode())
 
-def empty_spreadsheet_data():
+def empty_spreadsheet_data(lang):
     """Create an empty spreadsheet workbook.
     The sheet name should be the same for all users to allow consistent references
     in formulas. It is translated for the user creating the spreadsheet.
     """
+    locale = odooLangToSpreadsheetLocale(lang)
```
Comment (LucasLefevre) [inline addons/spreadsheet/utils.py:19]: use `snake_case` in python

File: addons/spreadsheet/utils.py [hunk 3]
```diff
@@ -1,26 +1,97 @@
...
+        spreadsheetSymbol = strftimeToSpreadsheetDateFormatTable.get(symbol)
+        if spreadsheetSymbol:
+            parts.append(spreadsheetSymbol)
+
+    separator = re.search(r"(/|-| )", strfFormat)
+    separator = separator.group(1) if separator else "/"
+
+    return separator.join(parts)
+
+def odooLangToSpreadsheetLocale(lang):
```
Comment (LucasLefevre) [inline addons/spreadsheet/utils.py:86]: make this a `res.lang` method and use dot notation `lang.code`

File: addons/spreadsheet/utils.py [hunk 4]
```diff
@@ -1,26 +1,97 @@
...
+    """Convert an odoo lang to a spreadsheet locale.
+    """
+    return {
+        "name": lang["name"],
+        "code":  re.sub("_", "-", lang["code"]),
```
Comment (LucasLefevre) [inline addons/spreadsheet/utils.py:91]: any reason we use "-" instead of "_" ?

File: addons/spreadsheet/models/res_lang.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+from odoo import api, models
+
+from odoo.addons.spreadsheet.utils import odooLangToSpreadsheetLocale
+
+
+class Lang(models.Model):
+    _inherit = "res.lang"
+
+    @api.model
+    def get_locales_for_spreadsheet(self):
```
Comment (LucasLefevre) [inline addons/spreadsheet/models/res_lang.py:12]: ? 
```suggestion
    def get_spreadsheet_locales(self):
```

Comment (hokolomopo) [reply addons/spreadsheet/models/res_lang.py:12]: What I didn't like about this name is that it kinda implies that we are getting the locales _of_ a spreadsheet ...

**Pull request closed without merging**